File: tests/expectations/include_all.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

enum UnusedEnum
#if __STDC_VERSION__ >= 202311L
  : uint8_t
#endif // __STDC_VERSION__ >= 202311L
 {
  A,
  B,
};
#if __STDC_VERSION__ >= 202311L
typedef enum UnusedEnum UnusedEnum;
#else
typedef uint8_t UnusedEnum;
#endif // __STDC_VERSION__ >= 202311L

typedef struct ExplicitOpaque ExplicitOpaque;

typedef struct OpaqueDependency OpaqueDependency;

typedef struct {
  int32_t x;
  float y;
} UnusedStruct;

typedef uint32_t UnusedTransparent;

typedef union {
  int32_t x;
  float y;
} UnusedUnion;

typedef int32_t UnusedAlias;

typedef struct {
  OpaqueDependency *opaque;
} UsesOpaqueDependency;

typedef struct {
  int32_t x;
} PublicInPrivateModule;

File: tests/rust/include_all.rs
#[repr(C)]
pub struct UnusedStruct {
    x: i32,
    y: f32,
}

#[repr(u8)]
pub enum UnusedEnum {
    A,
    B,
}

#[repr(transparent)]
pub struct UnusedTransparent(u32);

#[repr(C)]
pub union UnusedUnion {
    x: i32,
    y: f32,
}

pub type UnusedAlias = i32;

pub struct NotCbindgenable {
    x: i32,
}

pub struct ExplicitOpaque {
    x: i32,
}

struct OpaqueDependency {
    x: i32,
}

#[repr(C)]
pub struct UsesOpaqueDependency {
    opaque: *mut OpaqueDependency,
}

#[repr(C)]
pub struct ExcludedStruct {
    x: i32,
}

#[repr(C)]
struct PrivateStruct {
    x: i32,
}

#[repr(u8)]
enum PrivateEnum {
    A,
}

#[repr(transparent)]
struct PrivateTransparent(u32);

#[repr(C)]
union PrivateUnion {
    x: i32,
}

type PrivateAlias = i32;

mod private_module {
    #[repr(C)]
    pub struct PublicInPrivateModule {
        pub x: i32,
    }
}
